planner/core/find_best_task.go

```diff
@@ -423,7 +423,8 @@ func isCoveringIndex(columns []*expression.Column, indexColumns []*model.IndexCo
 		isIndexColumn := false
 		for _, indexCol := range indexColumns {
 			isFullLen := indexCol.Length == types.UnspecifiedLength || indexCol.Length == col.RetType.Flen
-			if col.ColName.L == indexCol.Name.L && isFullLen {
+			// We use col.OrigColName instead of col.ColName
```

Comment:
I think adding the issue link here can help others to understand.

Reply:
ok